File: src/com/nishyu/gui/RealTimeMarketDataRecord.java
package com.nishyu.gui;

public class RealTimeMarketDataRecord {
	private String _name;
	private double _price;
	private double _volume;
	
	public RealTimeMarketDataRecord() {
	}

	public RealTimeMarketDataRecord(String name, double price, double volume) {
		_name = name;
		_price = price;
		_volume = volume;
	}
	
	public String getName()  { return _name;   }
	public double getPrice() { return _price;  }
	public double getVolume(){ return _volume; }
	
	public void setPrice( double price )  { _price = price;  }
	public void setVolume( double volume ){ _volume = volume; }

	public String toString(){
		return String.format("%s %s: price = %f, volume = %f", this.getClass().toString(), getName(), getPrice(), getVolume() ); 
	}
}

File: core/src/main/scala/com/quantweb/mdserver/RealTimeMarketDataRecord.scala
package com.quantweb.mdserver
import org.apache.commons.lang3.SerializationUtils

@SerialVersionUID(-2644994007410833991L)
class RealTimeMarketDataRecord(name: String, price: Double, volume: Double ) extends java.io.Serializable {

    def getName = name
    def getPrice = price
    def getVolume = volume
    
    override def toString = "[RealTimeMarketDataRecord] %s: price = %f, volume = %f".format(name, price, volume) 
    
	def getBytes = SerializationUtils.serialize(this)

}
